### ISSUE TITLE
[v12] Position absolute never displayed onNodeChanges

### ISSUE THREAD

**Issue by CarolineBoyer** [xyflow/xyflow#4437]
### What platform were you using when you found the bug?

- React Flow / Svelte Flow version:  v12
- Browser and version: Chrome Version 126.0.6478.127 (Build officiel) (32 bits)
- OS and version: Windows 10 

### Describe the Bug

Not sure if it is missing from migration guide, or missing something obvious.

```typescript
onNodesChange={(changes) => {
  changes.forEach((oneChange) => {
    console.log(oneChange);
  }
}
```

it seems position object has changed : 
```json
{
    "id": "332bb62a-9181-468b-b70e-bb06d17c10b8",
    "type": "position",
    "position": {
        "x": null,
        "y": null
    },
    "dragging": true
}
```
positionAbsolute disappeared, and position x,y is often NaN/null.

This is probably an effect from other changes that are explained, but not sure which one and how to avoid that breaking change.

Cheers for this amazing lib !

### Steps to reproduce the bug or issue

1. Create blocks
2. Track onNodesChange function (in my case we have a collaborative sharing of position)
3. Try to move it around, observe

### Expected behavior

API used to have positionAbsolute that was perfect to track changes in position. 
The NaN of position makes it impossible to use that one =(

https://reactflow.dev/api-reference/types/node-change#nodepositionchange

**Comment by moklick** [comment]
Hey! Thanks for the report. 

The position should not be null. I will try to find out what is happening but it seems like a bug. We now distinguish between user nodes and internal nodes. If you need the absolute position, you can do:

`const internalNode = getInternalNode(id);`

Does that help for now?

**Comment by CarolineBoyer** [comment]
Hi ! So fast to answer, thanks.

Basically, I went and share 
```typescript
onNodesChange={(changes) => {
  changes.forEach((oneChange) => {
    if (
      oneChange.type === 'position' &&
      oneChange.dragging === true &&
      oneChange.position?.x &&
      typeof oneChange.position?.x === 'number' &&
      oneChange.position?.y &&
      typeof oneChange.position?.y === 'number'
    ) {
    //broadcast position change
    }
 }
```

and it does work until fixing :D 

I wouldn't be against an explanation on position vs positionAbsolute then. 
Position seems now to be the general x,y on the screen.
But not sure what PositionAbsolute is then ? the delta when moving around?

I used to use PositionAbsolute, as it was mimicking position change better, but your answer makes me think an update in docs https://reactflow.dev/api-reference/types/node-change#nodepositionchange would just be great for stupids (no offense to all of us out there) like me.

**Comment by moklick** [comment]
I can't reproduce this. How do you get null or NaN for the position?

**Comment by matt-yard** [comment]
I'm also getting NaN for position x,y after migrating to v12. For us it only seems to happen after we refetch the data from our server and overwrite the local state, for example:

- Fetch initial nodes and edges from the server and set them in local state
- We can drag the nodes around with no issues
- A background refresh fetches the nodes and edges again from the server and sets the data to local state, overwriting the current state
-Dragging a node now gives NaN for x and y position

I noticed that the every time we overwrite the state after a refetch, the "measured" property becomes undefined, and is never recalculated (not sure if this is a bug, or we're missing something with v12), but If the measured.height and measured.width properties are being used to calculate the x y position on drag, then maybe them being undefined is whats causing our NaN values?

I will try and recreate this in a code sandbox if i get the time later on, but I just thought I'd share our problems so far in case it helps.

**Comment by moklick** [comment]
Thanks @matt-yard I will try to find out what's wrong here.

**Comment by matt-yard** [comment]
@moklick We are using a custom onNodesChange. It just calls the applyNodeChanges function, and we haven't made any changes to it with v12. I was able to recreate our issue in a codesandbox here https://codesandbox.io/p/sandbox/dreamy-ride-s9dqtn

To reproduce:
- Drag a node and drop it
- Click the "Refetch" button to simulate a refresh of the data from a "server"
- Try moving any of the nodes after the refetch

**Comment by moklick** [comment]
This is fixed in 12.0.3 however if you send nodes to the server and then fetch nodes again it's important to keep the `measured` attribute so that you don't re-measure them if not necessary. `node.width` in v11 is now `node.measured.width`. If you have `node.width`/ `node.height` in your db, you could remove those attributes on the client before setting the nodes and then just store measured.
